### PR TITLE
Iteration 4 methods - Ryan

### PR DESCRIPTION
**Pull request by rallen20**
Summary:
All methods excluding seasonal methods are functional and passing. The specific methods completed in this PR include iteration 4 methods, biggest_team_blowout, worst_loss, and head_to_head.

Files Changes:
See Below

Tests:
All non-seasonal tests are passing
Rake: 92 runs, 177 assertions, 0 failures, 0 errors, 0 skips
Spec Harness: .......................FF........FFFFFFFFF

### CODE REVIEW

File: lib/game_collection.rb [hunk 1]
```diff
@@ -113,7 +113,6 @@ def games_by_season(team_id)
         game_season[:season] = game.season
         game_season[:away_team_id] = game.away_team_id
         game_season[:home_team_id] = game.home_team_id
-        require "pry"; binding.pry
```
Comment (LainMcGrathTuring) [inline lib/game_collection.rb:116]: This method should be deleted and not merged into master

File: test/game_collection_test.rb [hunk 2]
```diff
@@ -61,6 +61,6 @@ def test_it_can_calculate_average_goals_by_season
   end
 
   def test_it_can_find_games_by_season
-    assert_equal 12, @game_collection.games_by_season("26")
+    assert_equal 20, @game_collection.games_by_season("26").length
```
Comment (LainMcGrathTuring) [inline test/game_collection_test.rb]: This test should be deleted since this method is not yet functional

Comment (mintona) [inline test/game_collection_test.rb]: I think we actually want to comment out this test for now, not change the expected value to 20. It's my understanding that the expected value should be 12 and the method isn't done yet.

**Review by mintona — COMMENTED** [review]
Everything looks good except we need to check on the game_collection_test line 64 to see if that needs to be removed in this version.

**Comment by mintona** [comment]
All code in question has been removed for this pull request and will be worked on in another branch.

**Pull request merged**